Source/WebCore/accessibility/AccessibilityObject.cpp

```diff
@@ -3691,9 +3690,9 @@ void AccessibilityObject::scrollToMakeVisibleWithSubFocus(IntRect&& subfocus) co
     // Search up the parent chain until we find the first one that's scrollable.
     AccessibilityObject* scrollParent = parentObject();
     ScrollableArea* scrollableArea;
-    for (scrollableArea = nullptr;
-         scrollParent && !(scrollableArea = scrollParent->getScrollableAreaIfScrollable());
-         scrollParent = scrollParent->parentObject()) { }
+    for (scrollableArea = nullptr; scrollParent
+        && !(scrollableArea = scrollParent->getScrollableAreaIfScrollable());
+        scrollParent = scrollParent->parentObject()) { }
     if (!scrollableArea)
         return;
```

Comment:
Could we re-write this entirely? This feels like an overall unintuitive use of a for-loop. I'd go for this personally:

```cpp
ScrollableArea* scrollableArea = nullptr;
while (scrollParent) {
    scrollableArea = scrollParent->getScrollableAreaIfScrollable();
    if (scrollableArea)
        break;
    scrollParent = scrollParent->parentObject();
}
```

Longer, but so much easier to understand what's happening.